flocker/control/_model.py

```diff
@@ -301,9 +347,8 @@ def __invariant__(self):
         return (True, "")
 
     hostname = field(type=unicode, factory=unicode, mandatory=True)
-    applications = pset_field(Application)
-    manifestations = field(type=PMap, initial=pmap(), factory=pmap,
-                           mandatory=True)
+    applications = pset_field(Application, optional=True)
+    manifestations = pmap_field(unicode, Manifestation, optional=True)
```

Comment:
Do we actually want these changes to be applied to `Node` or do they actually belong on `NodeState`?  The goal is to let the dataset convergence agent send back just manifestations and the container convergence agent send back just applications, right?  `P2PNodeDeployer.discover_local_state` returns a `NodeState`, not a `Node`.